app/src/main/java/com/example/tasky/agenda_details/data/ImageCompressorImpl.kt

```diff
@@ -12,20 +12,24 @@ import java.io.ByteArrayOutputStream
 
 class ImageCompressorImpl(private val contentResolver: ContentResolver) :
     ImageCompressor {
-    override suspend fun compressImage(drawable: Drawable): ByteArray {
+    override suspend fun compressImage(drawable: Drawable, quality: Int): ByteArray {
         return withContext(Dispatchers.IO) {
-            val quality = 80
             val bitmap = (drawable as BitmapDrawable).bitmap
             val stream = ByteArrayOutputStream()
             bitmap.compress(Bitmap.CompressFormat.JPEG, quality, stream)
             stream.toByteArray()
         }
     }
 
-    override suspend fun uriToDrawable(uri: Uri): Drawable? {
+    override suspend fun uriStringToDrawable(uri: String): Drawable {
         return withContext(Dispatchers.IO) {
-            val inputStream = contentResolver.openInputStream(uri)
-            Drawable.createFromStream(inputStream, uri.toString())
+            val inputStream = contentResolver.openInputStream(Uri.parse(uri))
+            requireNotNull(inputStream) { "Failed to open input stream for URI: $uri" }
```

Comment:
I'd be very careful with non-null assertions. This will make your app crash of the Uri doesn't exist for some reason. Why not just have a simple null-check? Also I would recommend to use `.use` for the input stream since that closes it automatically after use (you keep it open forever here :D):
```
return contentResolver.openInputStream(Uri.parse(uri))?.use { inputStream -> 
    Drawable.createFromStream(inputStream, uri)
}
```